classes/class-kco-checkout.php

```diff
@@ -109,14 +109,17 @@ public function update_klarna_order() {
 
 		$klarna_order = KCO_WC()->api->get_klarna_order( $klarna_order_id );
 
-		if ( $klarna_order && 'checkout_incomplete' === $klarna_order['status'] ) {
+		// Store the status in case it is omitted from the response in the update request.
+		$status = $klarna_order['status'];
```

Comment:
I think you need to add a default here, rather then bellow. Since if status is not set here it will generate a warning message. Maybe set it to null as the default instead of a string value, and then just use the `$status` variable in all checks later.